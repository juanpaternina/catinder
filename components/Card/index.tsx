import { StyleSheet, Text, View } from "react-native";
import { Image } from "expo-image";

type Props = {
  image: string;
  name: string;
  age: number;
  origin: string;
};

export default function Card({ image, name, age, origin }: Props) {
  return (
    <View style={{ position: "absolute", width: "100%", height: "100%" }}>
      <View style={styles.container}>
        <Image
          testID="cat-image"
          contentFit="cover"
          //source={{ uri: "https://picsum.photos/343/440" }}
          source={{ uri: image }}
          style={styles.image}
        />
        <View style={styles.infoContainer}>
          <View style={styles.titleDescriptionContainer}>
            <Text testID="cat-name" style={styles.title}>
              {name}
            </Text>
            <Text testID="cat-age" style={styles.title}>
              {age}
            </Text>
          </View>
          <Text testID="cat-origin" style={styles.description}>
            {origin}
          </Text>
        </View>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    width: "100%",
    height: "100%",
    boxShadow: "0px 10px 16px 0px #BFBFC04D",
    borderRadius: 16,
    justifyContent: "flex-end",
    alignSelf: "center",
  },
  image: {
    width: "100%",
    height: "100%",
    borderRadius: 16,
    position: "absolute",
    backgroundColor: "#FFF",
  },
  infoContainer: {
    width: 307,
    height: 48,
    alignSelf: "center",
    backgroundColor: "#FFF",
    paddingHorizontal: 16,
    paddingTop: 8,
    borderTopEndRadius: 16,
    borderTopLeftRadius: 16,
  },
  titleDescriptionContainer: {
    flexDirection: "row",
    justifyContent: "space-between",
  },
  title: {
    fontWeight: "bold",
    fontSize: 16,
    lineHeight: 20,
  },
  description: {
    fontSize: 10,
    lineHeight: 10,
    fontWeight: "bold",
    color: "#BFBFC0",
    marginTop: 2,
  },
});
