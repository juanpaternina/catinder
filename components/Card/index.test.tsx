import { Text } from "react-native";
import { render, screen } from "@testing-library/react-native";
import Card from "./index";

describe("Card", () => {
  it("renders correctly", () => {
    render(
      <Card
        image="https://picsum.photos/343/440"
        name="Cat Name"
        age={6}
        origin="Egypt"
      />
    );
    expect(screen.getByTestId("cat-image")).toBeTruthy();
    expect(screen.getByTestId("cat-name")).toBeTruthy();
    expect(screen.getByTestId("cat-age")).toBeTruthy();
    expect(screen.getByTestId("cat-origin")).toBeTruthy();
    expect(screen.getByText("Cat Name")).toBeTruthy();
    expect(screen.getByText("6")).toBeTruthy();
    expect(screen.getByText("Egypt")).toBeTruthy();
    expect(screen).toMatchSnapshot();
  });
});
