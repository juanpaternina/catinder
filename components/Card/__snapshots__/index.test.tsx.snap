// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<View
  style={
    {
      "borderRadius": 16,
      "boxShadow": "0px 10px 16px 0px #BFBFC04D",
      "height": 440,
      "justifyContent": "flex-end",
      "width": 343,
    }
  }
>
  <ViewManagerAdapter_ExpoImage
    borderRadius={16}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height="100%"
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    position="absolute"
    source={
      [
        {
          "uri": "https://picsum.photos/343/440",
        },
      ]
    }
    style={
      {
        "borderRadius": 16,
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      }
    }
    testID="cat-image"
    transition={null}
    width="100%"
  />
  <View
    style={
      {
        "alignSelf": "center",
        "backgroundColor": "#FFF",
        "borderTopEndRadius": 16,
        "borderTopLeftRadius": 16,
        "height": 48,
        "paddingHorizontal": 16,
        "paddingTop": 8,
        "width": 307,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 16,
            "fontWeight": "bold",
            "lineHeight": 20,
          }
        }
        testID="cat-name"
      >
        Cat Name
      </Text>
      <Text
        style={
          {
            "fontSize": 16,
            "fontWeight": "bold",
            "lineHeight": 20,
          }
        }
        testID="cat-age"
      >
        6
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#BFBFC0",
          "fontSize": 10,
          "fontWeight": "bold",
          "lineHeight": 10,
          "marginTop": 2,
        }
      }
      testID="cat-origin"
    >
      Egypt
    </Text>
  </View>
</View>
`;
